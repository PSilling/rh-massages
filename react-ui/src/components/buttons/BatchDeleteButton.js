// react imports
import React, { Component } from "react";
import PropTypes from "prop-types";

// component imports
import ConfirmationModal from "../modals/ConfirmationModal";

// util imports
import _t from "../../util/Translations";

/**
 * A button used for batch Massage deletion. Contains a ConfirmationModal to confirm
 * the action.
 */
class BatchDeleteButton extends Component {
  state = { active: false };

  handleToggle = () => {
    this.setState(prevState => ({ active: !prevState.active }));
  };

  render() {
    return (
      <span>
        <button type="button" className="btn btn-default" onClick={this.handleToggle} disabled={this.props.disabled}>
          {this.props.label}
        </button>
        {this.state.active && (
          <ConfirmationModal
            message={_t.translate("Are you sure? This action cannot be reverted.")}
            onClose={this.handleToggle}
            onConfirm={() => {
              this.handleToggle();
              this.props.onDelete();
            }}
          />
        )}
      </span>
    );
  }
}

BatchDeleteButton.propTypes = {
  /** function to be called on action confirmation */
  onDelete: PropTypes.func.isRequired,
  /** whether the button should be disabled */
  disabled: PropTypes.bool,
  /** button label to be displayed */
  label: PropTypes.string
};

BatchDeleteButton.defaultProps = {
  disabled: false,
  label: "Delete"
};

export default BatchDeleteButton;
