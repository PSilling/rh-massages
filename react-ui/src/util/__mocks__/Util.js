const get = jest.fn((url, update) => {
  update([]);
});
const post = jest.fn((url, data, update) => {
  update();
});
const put = jest.fn((url, data, update) => {
  update();
});
const notify = jest.fn();
const clearAllIntervals = jest.fn();
const getEventLink = jest.fn();

const isEmpty = jest.fn(object => object === null || typeof object === "undefined" || object === "");

const findInArrayById = jest.fn((array, id) => {
  for (let i = 0; i < array.length; i++) {
    if (array[i].id === id) {
      return i;
    }
  }
  return -1;
});

let tooltipCount = 0;
const getTooltipTargets = jest.fn(count => {
  const targets = [];
  for (let i = 0; i < count; i++) {
    targets.push(`Tooltip${tooltipCount++}`);
  }
  return targets;
});

module.exports = {
  get,
  post,
  put,
  notify,
  clearAllIntervals,
  getEventLink,
  isEmpty,
  findInArrayById,
  tooltipCount,
  getTooltipTargets
};
