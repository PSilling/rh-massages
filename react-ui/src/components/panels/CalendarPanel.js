// react imports
import React, { Component } from "react";
import PropTypes from "prop-types";

// module imports
import { Row, Col } from "reactstrap";
import moment from "moment";
import BigCalendar from "react-big-calendar";

// component imports
import CalendarToolbar from "../util/CalendarToolbar";
import MassageEvent from "./components/MassageEvent";
import MassageEventModal from "../modals/MassageEventModal";
import WeekdayHeader from "./components/WeekdayHeader";

// util imports
import Auth from "../../util/Auth";
import _t from "../../util/Translations";
import Util from "../../util/Util";

const localizer = BigCalendar.momentLocalizer(moment);

/**
 * Calendar with massage events. Supports color feedback and Massage administration.
 */
class CalendarPanel extends Component {
  state = {
    active: false,
    action: "",
    selectedEvent: null,
    label: "",
    overtime: false,
    date: new Date(),
    view: this.props.allowEditation ? "work_week" : "month"
  };

  localization = {
    allDay: _t.translate("All day"),
    date: _t.translate("Date"),
    time: _t.translate("Time"),
    event: _t.translate("Event"),
    showMore: total => `+ ${_t.translate("Show more")} (${total})`
  };

  formats = {
    dayFormat: (date, culture, local) => local.format(date, _t.translate("DD ddd"), culture),
    eventTimeRangeFormat: () => null
  };

  doubleClick = false;

  eventStyler = event => ({
    style: {
      backgroundColor: event.bgColor,
      color: "black",
      border: 0,
      borderRadius: "4px",
      marginLeft: this.state.view === "month" ? "0px" : "5px",
      opacity: Util.findInArrayById(this.props.selected, event.massage.id) !== -1 ? "0.8" : "1.0"
    }
  });

  onSelectEvent = event => {
    if (this.props.selectEvents) {
      this.props.onSelect(event);
    } else if (this.props.allowEditation) {
      this.props.onTooltipTrigger(`MassageEventID${event.massage.id}`);
    } else {
      this.setState({ active: true, action: "none", label: "none", selectedEvent: event });
      this.props.onTooltipTrigger(null);
    }
  };

  onEventDoubleClick = event => {
    if (this.props.allowEditation) {
      this.configureModalActions(event);
    } else {
      this.setState({ active: true, action: "none", label: "none", selectedEvent: event });
    }
    this.props.onTooltipTrigger(null);
  };

  handleEventSelect = event => {
    setTimeout(() => {
      this.doubleClick = false;
    }, 500);

    if (this.doubleClick) {
      this.onEventDoubleClick(event);
    }

    this.onSelectEvent(event);
    this.doubleClick = !this.doubleClick;
  };

  /**
   * Configures MassageEventModal buttons based on event client assignment status.
   */
  configureModalActions = event => {
    if (Util.isEmpty(event.massage.client)) {
      if (
        this.props.massageMinutes + moment(event.massage.ending).diff(moment(event.massage.date), "minutes") >
        Util.MAX_MASSAGE_MINS
      ) {
        this.setState({
          active: true,
          action: "none",
          selectedEvent: event,
          label: _t.translate("Assign me"),
          overtime: true
        });
      } else {
        this.setState({
          active: true,
          action: "assign",
          selectedEvent: event,
          label: _t.translate("Assign me"),
          overtime: false
        });
      }
    } else if (Auth.getSub() === event.massage.client.sub) {
      this.setState({
        active: true,
        action: "cancel",
        selectedEvent: event,
        label: _t.translate("Unassign me"),
        overtime: false
      });
    } else if (Auth.isAdmin() || (Auth.isMasseur() && Auth.getSub() === event.massage.masseuse.sub)) {
      this.setState({
        active: true,
        action: "cancel",
        selectedEvent: event,
        label: _t.translate("Force cancel"),
        overtime: false
      });
    } else {
      this.setState({
        active: true,
        action: "none",
        selectedEvent: event,
        label: "none",
        overtime: false
      });
    }
  };

  editEvent = () => {
    this.props.onEdit(this.state.selectedEvent.massage);
    this.setState({ active: false });
  };

  deleteEvent = () => {
    this.props.onDelete(this.state.selectedEvent.massage.id);
    this.setState({ active: false });
  };

  handleToggle = () => {
    this.setState(prevState => ({ active: !prevState.active }));
    this.props.onTooltipTrigger(null);
  };

  changeView = view => {
    this.setState({ view });
    this.props.onDateChange(this.state.date, view);
  };

  changeDate = left => {
    this.setState(prevState => {
      const dateAsMoment = moment(prevState.date);
      const modifier = prevState.view === "month" ? "month" : "week";

      if (left) {
        dateAsMoment.subtract(1, modifier);
      } else {
        dateAsMoment.add(1, modifier);
      }

      const newDate = dateAsMoment.toDate();
      this.props.onDateChange(newDate, prevState.view);
      return { date: newDate };
    });
  };

  render() {
    return (
      <div className="mb-3">
        <Row>
          <Col md="12">
            <CalendarToolbar
              month={
                this.state.view === "month"
                  ? moment(this.state.date).format("MMMM YYYY")
                  : moment(this.state.date)
                      .subtract(2, "days")
                      .format("MMMM YYYY")
              }
              monthActive={this.state.view === "month"}
              leftDisabled={this.props.allowEditation && moment(this.state.date).isBefore(moment())}
              rightDisabled={
                !this.props.allowEditation &&
                moment(this.state.date)
                  .add(1, "day")
                  .isAfter(moment())
              }
              leftAction={() => this.changeDate(true)}
              rightAction={() => this.changeDate(false)}
              onViewChange={this.changeView}
            />
            <BigCalendar
              messages={this.localization}
              date={this.state.date}
              events={this.props.events}
              onView={view => this.props.onDateChange(moment(this.state.date), view)}
              onNavigate={() => {}}
              view={this.state.view}
              views={["work_week", "month"]}
              style={{ height: "85vh" }}
              timeslots={1}
              eventPropGetter={this.eventStyler}
              onSelectEvent={this.handleEventSelect}
              startAccessor={event => new Date(event.massage.date)}
              endAccessor={event => new Date(event.massage.ending)}
              selectable={Auth.isAdminOrMasseur() && this.props.allowEditation && this.state.view === "work_week"}
              onSelectSlot={this.props.onAdd}
              min={new Date("2018-01-01T08:30:00")}
              max={new Date("2018-01-01T18:00:00")}
              popup
              toolbar={false}
              localizer={localizer}
              formats={this.formats}
              components={{
                event: props => (
                  <MassageEvent
                    event={props.event}
                    archived={!this.props.allowEditation}
                    activeTooltip={this.props.activeEventTooltip}
                    massageMinutes={this.props.massageMinutes}
                    onAssign={this.props.onAssign}
                    onCancel={this.props.onCancel}
                    onDelete={this.props.onDelete}
                    onEdit={this.props.onEdit}
                    onShowMore={this.onEventDoubleClick}
                    view={this.state.view}
                  />
                ),
                work_week: {
                  header: props =>
                    WeekdayHeader(props, Auth.isAdminOrMasseur() && this.props.selectEvents, this.props.onSelectDay)
                }
              }}
            />
          </Col>
        </Row>

        <Row className="text-center mt-3">
          <Col md="12">
            <strong>{_t.translate("Legend:")}</strong>
            <span
              style={{ backgroundColor: Util.SUCCESS_COLOR, borderRadius: "4px", padding: "8px", marginLeft: "8px" }}
            >
              {_t.translate("Free massage")}
            </span>
            {this.props.allowEditation && (
              <span
                style={{ backgroundColor: Util.WARNING_COLOR, borderRadius: "4px", padding: "8px", marginLeft: "8px" }}
              >
                {_t.translate("My massage")}
              </span>
            )}
            <span style={{ backgroundColor: Util.ERROR_COLOR, borderRadius: "4px", padding: "8px", marginLeft: "8px" }}>
              {_t.translate("Assigned massage")}
            </span>
          </Col>
        </Row>

        {this.state.active && (
          <MassageEventModal
            event={this.state.selectedEvent}
            label={this.state.label}
            disabled={this.state.overtime}
            onClose={this.handleToggle}
            onEdit={this.editEvent}
            onDelete={this.deleteEvent}
            allowEditation={this.props.allowEditation}
            onConfirm={() => {
              this.handleToggle();
              switch (this.state.action) {
                case "assign":
                  this.props.onAssign(this.state.selectedEvent.massage);
                  break;
                case "cancel":
                  this.props.onCancel(this.state.selectedEvent.massage);
                  break;
                case "none":
                  break;
                default:
                  Util.notify("error", "", _t.translate("An error occured!"));
              }
              this.setState({ selectedEvent: null });
            }}
          />
        )}
      </div>
    );
  }
}

CalendarPanel.propTypes = {
  /** events featured in the calendar */
  events: PropTypes.arrayOf(
    PropTypes.shape({
      bgColor: PropTypes.string,
      massage: PropTypes.shape({
        id: PropTypes.number,
        masseuse: PropTypes.shape({
          email: PropTypes.string,
          masseur: PropTypes.bool,
          name: PropTypes.string,
          sub: PropTypes.string,
          subscribed: PropTypes.bool,
          surname: PropTypes.string
        }),
        date: PropTypes.oneOfType([PropTypes.number, PropTypes.instanceOf(Date)]),
        ending: PropTypes.oneOfType([PropTypes.number, PropTypes.instanceOf(Date)]),
        client: PropTypes.shape({
          email: PropTypes.string,
          name: PropTypes.string,
          surname: PropTypes.string,
          sub: PropTypes.string,
          subscribed: PropTypes.bool
        }),
        facility: PropTypes.shape({
          id: PropTypes.number,
          name: PropTypes.string
        })
      })
    })
  ).isRequired,
  /** function called on view change */
  onDateChange: PropTypes.func.isRequired,
  /** function called on event deletion */
  onDelete: PropTypes.func.isRequired,
  /** function called on event selection */
  onSelect: PropTypes.func.isRequired,
  /** function called on full day event selection */
  onSelectDay: PropTypes.func.isRequired,
  /** all currently selected events */
  selected: PropTypes.arrayOf(
    PropTypes.shape({
      bgColor: PropTypes.string,
      massage: PropTypes.shape({
        id: PropTypes.number,
        masseuse: PropTypes.shape({
          email: PropTypes.string,
          masseur: PropTypes.bool,
          name: PropTypes.string,
          sub: PropTypes.string,
          subscribed: PropTypes.bool,
          surname: PropTypes.string
        }),
        date: PropTypes.oneOfType([PropTypes.number, PropTypes.instanceOf(Date)]),
        ending: PropTypes.oneOfType([PropTypes.number, PropTypes.instanceOf(Date)]),
        client: PropTypes.shape({
          email: PropTypes.string,
          name: PropTypes.string,
          surname: PropTypes.string,
          sub: PropTypes.string,
          subscribed: PropTypes.bool
        }),
        facility: PropTypes.shape({
          id: PropTypes.number,
          name: PropTypes.string
        })
      })
    })
  ).isRequired,
  /** whether multi event selection should be activated */
  selectEvents: PropTypes.bool.isRequired,
  /** currently active event tooltip target */
  activeEventTooltip: PropTypes.string,
  /** whether the edit button should be shown (Admin only) */
  allowEditation: PropTypes.bool,
  /** number of currently used Massage time in minutes */
  massageMinutes: PropTypes.number,
  /** function called on selected slot event addition */
  onAdd: PropTypes.func,
  /** function called on event assignment */
  onAssign: PropTypes.func,
  /** function called on event cancellation */
  onCancel: PropTypes.func,
  /** function called on event editation */
  onEdit: PropTypes.func,
  /** function called on event tooltip toggle trigger */
  onTooltipTrigger: PropTypes.func
};

CalendarPanel.defaultProps = {
  activeEventTooltip: null,
  allowEditation: true,
  massageMinutes: 0,
  onAdd() {},
  onAssign() {},
  onCancel() {},
  onEdit() {},
  onTooltipTrigger() {}
};

export default CalendarPanel;
