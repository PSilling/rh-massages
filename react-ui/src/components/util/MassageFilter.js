// react imports
import React, { Component } from 'react';
import PropTypes from 'prop-types';

// util imports
import Util from '../../util/Util';
import _t from '../../util/Translations';

/**
 * Custom Masssage filtering component (search input with status checkbox).
 */
class MassageFilter extends Component {

  render() {
    return (
      <div className="row">
        <div className="col-md-5">
          <div className="input-group" style={{ 'marginBottom': '15px' }}>
              <span className="input-group-addon"><span className="glyphicon glyphicon-search"></span></span>
              <input type="search" value={this.props.value} className="form-control"
                placeholder={ _t.translate('Search') } onChange={this.props.onSearchChange}
                autoFocus onFocus={Util.moveCursorToEnd}
              />
          </div>
        </div>

        <div className="form-group col-md-5">
          <div className="col-md-5 col-md-offset-1">
            <input id="fromInput" value={this.props.from}
              onChange={this.props.onFromChange} className="form-control"
              type="date" placeholder={ _t.translate('From') }
            />
          </div>

          <div className="col-md-1" style={{ 'marginTop': '6px' }}>
            <label htmlFor="toInput">{ "â€“" }</label>
          </div>

          <div className="col-md-5">
            <input id="toInput" value={this.props.to}
              onChange={this.props.onToChange} className="form-control"
              type="date" placeholder={ _t.translate('To') }
            />
          </div>
        </div>

        <div className="form-group col-md-2" style={{ 'marginTop': '6px' }}>
          <label className="checkbox-inline">
            <input type="checkbox" onChange={this.props.onCheck} checked={this.props.checked} />
            <strong>{ _t.translate('only free massages') }</strong>
          </label>
        </div>
      </div>
    );
  }
}

MassageFilter.propTypes = {
  value: PropTypes.string, // current value of the search input
  from: PropTypes.string, // current value of the from date input
  to: PropTypes.string, // current value of the to date input
  checked: PropTypes.bool, // whether the free Massages only checkbox should be checked
  onCheck: PropTypes.func, // function called on free Massages only checkbox change
  onFromChange: PropTypes.func, // function called on from date value change
  onToChange: PropTypes.func, // function called on to date value change
  onSearchChange: PropTypes.func // function called on search value change
};

export default MassageFilter
