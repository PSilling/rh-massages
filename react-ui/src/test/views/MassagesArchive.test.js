// react imports
import React from 'react';

// test imports
import MassagesArchive from '../../views/MassagesArchive';
import BatchDeleteButton from '../../components/buttons/BatchDeleteButton';
import CalendarPanel from '../../components/panels/CalendarPanel';
import MassageFilter from '../../components/util/MassageFilter';
import moment from 'moment';
import Util from '../../util/Util';

// test mocks
jest.mock('../../util/Auth');
jest.mock('../../util/Util');

afterAll(() => {
  jest.resetAllMocks();
});

test('renders content correctly', () => {
  const wrapper = shallow(<MassagesArchive />);

  expect(wrapper).toMatchSnapshot();
});

test('properly changes state variables', () => {
  Util.delete = jest.fn((url, update, notify = true) => { update(); });
  const testMoment = moment().add(1, 'days'),
        testMassages = [{
          id: 1,
          date: new Date(0),
          ending: new Date(1000),
          client: null,
          facility: { id: 1, name: "test" }
        }],
        wrapper = shallow(<MassagesArchive />);

  let filter = wrapper.find(MassageFilter),
      buttons = wrapper.find(BatchDeleteButton),
      panel = wrapper.find(CalendarPanel);

  wrapper.instance().setState({ loading: true });
  wrapper.instance().updateEvents(testMassages);
  expect(wrapper.instance().state.events[0].massage).toBe(testMassages[0]);
  expect(wrapper.instance().state.loading).toBe(false);

  filter.props().onFreeCheck({ target: { checked: true } });
  expect(wrapper.instance().state.freeOnly).toBe(true);
  filter.props().onSelectCheck({ target: { checked: true } });
  expect(wrapper.instance().state.selectEvents).toBe(true);
  filter.props().onFilterChange({ target: { value: "test" } });
  expect(wrapper.instance().state.search).toBe("test");

  expect(buttons.get(0).props.disabled).toBe(true);
  buttons.get(0).props.onDelete();
  expect(Util.delete).toHaveBeenCalledTimes(1);
  buttons.get(1).props.onDelete();
  expect(Util.get).toHaveBeenCalledTimes(4);
  expect(Util.delete).toHaveBeenCalledTimes(1);
  expect(panel.props().selected).toEqual([]);
  expect(panel.props().allowEditation).toBe(false);
  panel.props().onDelete(1);
  expect(Util.delete).toHaveBeenCalledTimes(2);

  panel.props().onDateChange(testMoment.clone(), 'month');
  expect(wrapper.instance().state.from).toEqual(
    testMoment.clone().startOf('month').subtract(37, 'days')
  );
  expect(wrapper.instance().state.to).toEqual(
    testMoment.clone().endOf('month').add(37, 'days')
  );
  panel.props().onDateChange(testMoment.clone(), '');
  expect(wrapper.instance().state.from).toEqual(
    testMoment.clone().startOf('isoWeek').subtract(7, 'days')
  );
  expect(wrapper.instance().state.to).toEqual(
    testMoment.endOf('isoWeek').add(5, 'days')
  );
  expect(wrapper.instance().state.loading).toBe(true);
  expect(wrapper.instance().state.selected).toEqual([]);

  panel.props().onSelect(wrapper.instance().state.events[0]);
  expect(wrapper.instance().state.selected[0]).toBe(
    wrapper.instance().state.events[0].massage
  );
  panel.props().onSelect(null);
  expect(wrapper.instance().state.selected).toEqual([]);
});
