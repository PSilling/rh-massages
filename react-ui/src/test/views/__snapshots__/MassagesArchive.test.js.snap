// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders content correctly 1`] = `
ShallowWrapper {
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <MassagesArchive />,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateError": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "host",
    "props": Object {
      "children": Array [
        <InfoAlert
          onClose={[Function]}
        >
          Na této stránce můžete prohlížet dokončené, archivované masáže.
        </InfoAlert>,
        <h1>
          <div
            className="loader float-right"
          />
          Archiv masáží
        </h1>,
        <hr />,
        <Row
          tag="div"
        >
          <Col
            className="text-right"
            md="12"
            tag="div"
            widths={
              Array [
                "xs",
                "sm",
                "md",
                "lg",
                "xl",
              ]
            }
          >
            <TooltipButton
              active={false}
              disabled={false}
              label="Vybírat"
              onClick={[Function]}
              tooltip="Výběr více masáží pro hromadné operace"
            />
            <ConfirmationButton
              className="mx-2"
              dialogMessage="Jste si jisti? Tato akce nemůže být navrácena."
              disabled={true}
              label="Smazat vybrané"
              onConfirm={[Function]}
              tooltip="Smazat vybrané masáže"
            />
            <ConfirmationButton
              dialogMessage="Jste si jisti? Tato akce nemůže být navrácena."
              disabled={false}
              label="Smazat vše"
              onConfirm={[Function]}
              tooltip="Vyčistit historii masáží"
            />
          </Col>
        </Row>,
        <CalendarPanel
          activeEventTooltip={null}
          allowEditation={false}
          events={Array []}
          massageMinutes={0}
          onAdd={[Function]}
          onAssign={[Function]}
          onCancel={[Function]}
          onDateChange={[Function]}
          onDelete={[Function]}
          onEdit={[Function]}
          onSelect={[Function]}
          onSelectDay={[Function]}
          onTooltipTrigger={[Function]}
          selectEvents={false}
          selected={Array []}
        />,
      ],
      "className": "my-3",
    },
    "ref": null,
    "rendered": Array [
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "function",
        "props": Object {
          "children": "Na této stránce můžete prohlížet dokončené, archivované masáže.",
          "onClose": [Function],
        },
        "ref": null,
        "rendered": "Na této stránce můžete prohlížet dokončené, archivované masáže.",
        "type": [Function],
      },
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": Array [
            <div
              className="loader float-right"
            />,
            "Archiv masáží",
          ],
        },
        "ref": null,
        "rendered": Array [
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "host",
            "props": Object {
              "className": "loader float-right",
            },
            "ref": null,
            "rendered": null,
            "type": "div",
          },
          "Archiv masáží",
        ],
        "type": "h1",
      },
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {},
        "ref": null,
        "rendered": null,
        "type": "hr",
      },
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "function",
        "props": Object {
          "children": <Col
            className="text-right"
            md="12"
            tag="div"
            widths={
              Array [
                "xs",
                "sm",
                "md",
                "lg",
                "xl",
              ]
            }
          >
            <TooltipButton
              active={false}
              disabled={false}
              label="Vybírat"
              onClick={[Function]}
              tooltip="Výběr více masáží pro hromadné operace"
            />
            <ConfirmationButton
              className="mx-2"
              dialogMessage="Jste si jisti? Tato akce nemůže být navrácena."
              disabled={true}
              label="Smazat vybrané"
              onConfirm={[Function]}
              tooltip="Smazat vybrané masáže"
            />
            <ConfirmationButton
              dialogMessage="Jste si jisti? Tato akce nemůže být navrácena."
              disabled={false}
              label="Smazat vše"
              onConfirm={[Function]}
              tooltip="Vyčistit historii masáží"
            />
          </Col>,
          "tag": "div",
        },
        "ref": null,
        "rendered": Object {
          "instance": null,
          "key": undefined,
          "nodeType": "function",
          "props": Object {
            "children": Array [
              <TooltipButton
                active={false}
                disabled={false}
                label="Vybírat"
                onClick={[Function]}
                tooltip="Výběr více masáží pro hromadné operace"
              />,
              <ConfirmationButton
                className="mx-2"
                dialogMessage="Jste si jisti? Tato akce nemůže být navrácena."
                disabled={true}
                label="Smazat vybrané"
                onConfirm={[Function]}
                tooltip="Smazat vybrané masáže"
              />,
              <ConfirmationButton
                dialogMessage="Jste si jisti? Tato akce nemůže být navrácena."
                disabled={false}
                label="Smazat vše"
                onConfirm={[Function]}
                tooltip="Vyčistit historii masáží"
              />,
            ],
            "className": "text-right",
            "md": "12",
            "tag": "div",
            "widths": Array [
              "xs",
              "sm",
              "md",
              "lg",
              "xl",
            ],
          },
          "ref": null,
          "rendered": Array [
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "class",
              "props": Object {
                "active": false,
                "disabled": false,
                "label": "Vybírat",
                "onClick": [Function],
                "tooltip": "Výběr více masáží pro hromadné operace",
              },
              "ref": null,
              "rendered": null,
              "type": [Function],
            },
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "class",
              "props": Object {
                "className": "mx-2",
                "dialogMessage": "Jste si jisti? Tato akce nemůže být navrácena.",
                "disabled": true,
                "label": "Smazat vybrané",
                "onConfirm": [Function],
                "tooltip": "Smazat vybrané masáže",
              },
              "ref": null,
              "rendered": null,
              "type": [Function],
            },
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "class",
              "props": Object {
                "dialogMessage": "Jste si jisti? Tato akce nemůže být navrácena.",
                "disabled": false,
                "label": "Smazat vše",
                "onConfirm": [Function],
                "tooltip": "Vyčistit historii masáží",
              },
              "ref": null,
              "rendered": null,
              "type": [Function],
            },
          ],
          "type": [Function],
        },
        "type": [Function],
      },
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "class",
        "props": Object {
          "activeEventTooltip": null,
          "allowEditation": false,
          "events": Array [],
          "massageMinutes": 0,
          "onAdd": [Function],
          "onAssign": [Function],
          "onCancel": [Function],
          "onDateChange": [Function],
          "onDelete": [Function],
          "onEdit": [Function],
          "onSelect": [Function],
          "onSelectDay": [Function],
          "onTooltipTrigger": [Function],
          "selectEvents": false,
          "selected": Array [],
        },
        "ref": null,
        "rendered": null,
        "type": [Function],
      },
    ],
    "type": "div",
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "host",
      "props": Object {
        "children": Array [
          <InfoAlert
            onClose={[Function]}
          >
            Na této stránce můžete prohlížet dokončené, archivované masáže.
          </InfoAlert>,
          <h1>
            <div
              className="loader float-right"
            />
            Archiv masáží
          </h1>,
          <hr />,
          <Row
            tag="div"
          >
            <Col
              className="text-right"
              md="12"
              tag="div"
              widths={
                Array [
                  "xs",
                  "sm",
                  "md",
                  "lg",
                  "xl",
                ]
              }
            >
              <TooltipButton
                active={false}
                disabled={false}
                label="Vybírat"
                onClick={[Function]}
                tooltip="Výběr více masáží pro hromadné operace"
              />
              <ConfirmationButton
                className="mx-2"
                dialogMessage="Jste si jisti? Tato akce nemůže být navrácena."
                disabled={true}
                label="Smazat vybrané"
                onConfirm={[Function]}
                tooltip="Smazat vybrané masáže"
              />
              <ConfirmationButton
                dialogMessage="Jste si jisti? Tato akce nemůže být navrácena."
                disabled={false}
                label="Smazat vše"
                onConfirm={[Function]}
                tooltip="Vyčistit historii masáží"
              />
            </Col>
          </Row>,
          <CalendarPanel
            activeEventTooltip={null}
            allowEditation={false}
            events={Array []}
            massageMinutes={0}
            onAdd={[Function]}
            onAssign={[Function]}
            onCancel={[Function]}
            onDateChange={[Function]}
            onDelete={[Function]}
            onEdit={[Function]}
            onSelect={[Function]}
            onSelectDay={[Function]}
            onTooltipTrigger={[Function]}
            selectEvents={false}
            selected={Array []}
          />,
        ],
        "className": "my-3",
      },
      "ref": null,
      "rendered": Array [
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "function",
          "props": Object {
            "children": "Na této stránce můžete prohlížet dokončené, archivované masáže.",
            "onClose": [Function],
          },
          "ref": null,
          "rendered": "Na této stránce můžete prohlížet dokončené, archivované masáže.",
          "type": [Function],
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": Array [
              <div
                className="loader float-right"
              />,
              "Archiv masáží",
            ],
          },
          "ref": null,
          "rendered": Array [
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "host",
              "props": Object {
                "className": "loader float-right",
              },
              "ref": null,
              "rendered": null,
              "type": "div",
            },
            "Archiv masáží",
          ],
          "type": "h1",
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {},
          "ref": null,
          "rendered": null,
          "type": "hr",
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "function",
          "props": Object {
            "children": <Col
              className="text-right"
              md="12"
              tag="div"
              widths={
                Array [
                  "xs",
                  "sm",
                  "md",
                  "lg",
                  "xl",
                ]
              }
            >
              <TooltipButton
                active={false}
                disabled={false}
                label="Vybírat"
                onClick={[Function]}
                tooltip="Výběr více masáží pro hromadné operace"
              />
              <ConfirmationButton
                className="mx-2"
                dialogMessage="Jste si jisti? Tato akce nemůže být navrácena."
                disabled={true}
                label="Smazat vybrané"
                onConfirm={[Function]}
                tooltip="Smazat vybrané masáže"
              />
              <ConfirmationButton
                dialogMessage="Jste si jisti? Tato akce nemůže být navrácena."
                disabled={false}
                label="Smazat vše"
                onConfirm={[Function]}
                tooltip="Vyčistit historii masáží"
              />
            </Col>,
            "tag": "div",
          },
          "ref": null,
          "rendered": Object {
            "instance": null,
            "key": undefined,
            "nodeType": "function",
            "props": Object {
              "children": Array [
                <TooltipButton
                  active={false}
                  disabled={false}
                  label="Vybírat"
                  onClick={[Function]}
                  tooltip="Výběr více masáží pro hromadné operace"
                />,
                <ConfirmationButton
                  className="mx-2"
                  dialogMessage="Jste si jisti? Tato akce nemůže být navrácena."
                  disabled={true}
                  label="Smazat vybrané"
                  onConfirm={[Function]}
                  tooltip="Smazat vybrané masáže"
                />,
                <ConfirmationButton
                  dialogMessage="Jste si jisti? Tato akce nemůže být navrácena."
                  disabled={false}
                  label="Smazat vše"
                  onConfirm={[Function]}
                  tooltip="Vyčistit historii masáží"
                />,
              ],
              "className": "text-right",
              "md": "12",
              "tag": "div",
              "widths": Array [
                "xs",
                "sm",
                "md",
                "lg",
                "xl",
              ],
            },
            "ref": null,
            "rendered": Array [
              Object {
                "instance": null,
                "key": undefined,
                "nodeType": "class",
                "props": Object {
                  "active": false,
                  "disabled": false,
                  "label": "Vybírat",
                  "onClick": [Function],
                  "tooltip": "Výběr více masáží pro hromadné operace",
                },
                "ref": null,
                "rendered": null,
                "type": [Function],
              },
              Object {
                "instance": null,
                "key": undefined,
                "nodeType": "class",
                "props": Object {
                  "className": "mx-2",
                  "dialogMessage": "Jste si jisti? Tato akce nemůže být navrácena.",
                  "disabled": true,
                  "label": "Smazat vybrané",
                  "onConfirm": [Function],
                  "tooltip": "Smazat vybrané masáže",
                },
                "ref": null,
                "rendered": null,
                "type": [Function],
              },
              Object {
                "instance": null,
                "key": undefined,
                "nodeType": "class",
                "props": Object {
                  "dialogMessage": "Jste si jisti? Tato akce nemůže být navrácena.",
                  "disabled": false,
                  "label": "Smazat vše",
                  "onConfirm": [Function],
                  "tooltip": "Vyčistit historii masáží",
                },
                "ref": null,
                "rendered": null,
                "type": [Function],
              },
            ],
            "type": [Function],
          },
          "type": [Function],
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "class",
          "props": Object {
            "activeEventTooltip": null,
            "allowEditation": false,
            "events": Array [],
            "massageMinutes": 0,
            "onAdd": [Function],
            "onAssign": [Function],
            "onCancel": [Function],
            "onDateChange": [Function],
            "onDelete": [Function],
            "onEdit": [Function],
            "onSelect": [Function],
            "onSelectDay": [Function],
            "onTooltipTrigger": [Function],
            "selectEvents": false,
            "selected": Array [],
          },
          "ref": null,
          "rendered": null,
          "type": [Function],
        },
      ],
      "type": "div",
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
        "lifecycles": Object {
          "componentDidUpdate": Object {
            "onSetState": true,
          },
          "getDerivedStateFromProps": true,
          "getSnapshotBeforeUpdate": true,
          "setState": Object {
            "skipsComponentDidUpdateOnNullish": true,
          },
        },
      },
    },
  },
}
`;
